File: 283-move-zeroes/move-zeroes.java
class Solution {

    public void swap(int[] nums , int a , int b ){
        int temp = nums[a];
        nums[a] = nums[b]; 
        nums[b]  = temp ;
    }


    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int i = 0 ;
        int j =0 ;
        while(j<n){
            while(j<n && nums[j]==0){
                j++;
            }
            if(j==n){
                break;
            }
            swap(nums,i, j);
            i++;
            j++;
        }
    }
}

File: 283-move-zeroes/move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& a) {
        int n = a.size();
		int i =0 ; 
        int j = 0;
        while(j<n){
            if(a[i]!=0){
                i++;
                j++;
            }else{
                if(a[j]!=0){
                    int temp  = a[i];
                    a[i] = a[j];
                    a[j] = temp;
                    i++;
                }else{
                    j++;
                }
            }
        }
    }
};